### PR TITLE
A simple class for mpos

### PR DESCRIPTION
**Pull request by mganahl**
depends #578

### CODE REVIEW

File: tensornetwork/matrixproductstates/mpo.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    super().__init__(tensors=mpo, backend=backend, name=name)
+
+
+class FiniteTFI(FiniteMPO):
+  """
+  The famous transverse field Ising Hamiltonian. Everyone loves it
+  because any method looks great on it. It takes about the same time 
+  to solve it on an abacus as it takes on a NISQ device (the error bars on 
+  the latter are big though).
+  My 4 months old daughter today derived that the ground state energy of
```
[the remark this answers is not in the document]
Comment (mganahl) [reply tensornetwork/matrixproductstates/mpo.py:217]: lol, that's how trivial the ising model is. I'll remove it!

File: tensornetwork/matrixproductstates/mpo.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+class FiniteMPO(BaseMPO):
+  """
+  Base class for implementation of finite MPOs. Users should implement 
+  specific finite MPOs by deriving from FiniteMPO
+  """
+
+  def __init__(self,
+               tensors: List[Tensor],
+               backend: Optional[Union[BaseBackend, Text]] = None,
+               name: Optional[Text] = None) -> None:
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/mpo.py:106]: Document

File: tensornetwork/matrixproductstates/mpo.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+class InfiniteMPO(BaseMPO):
+  """
+  Base class for implementation of infinite MPOs. Users should implement 
+  specific infinite MPOs by deriving from InfiniteMPO.
+  """
+
+  def __init__(self,
+               tensors: List[Tensor],
+               backend: Optional[Union[BaseBackend, Text]] = None,
+               name: Optional[Text] = None) -> None:
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/mpo.py:85]: Document

File: tensornetwork/matrixproductstates/mpo.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    """
+    Initialize a BaseMPO.
+    Args:
+      tensors: A list of `Tensor` objects.
+      backend: The name of the backend that should be used to perform 
+        contractions.
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/mpo.py:39]: `name` is missing.

**Pull request merged**